src/Core/Content/Media/Thumbnail/ThumbnailService.php

```diff
@@ -345,52 +345,16 @@ private function calculateThumbnailSize(
         MediaThumbnailSizeEntity $preferredThumbnailSize,
         MediaFolderConfigurationEntity $config
     ): array {
-        if (!$config->getKeepAspectRatio()) {
-            $calculatedWidth = $preferredThumbnailSize->getWidth();
-            $calculatedHeight = $preferredThumbnailSize->getHeight();
-
-            $useOriginalSizeInThumbnails = $imageSize['width'] < $calculatedWidth || $imageSize['height'] < $calculatedHeight;
-
-            return $useOriginalSizeInThumbnails ? [
-                'width' => $imageSize['width'],
-                'height' => $imageSize['height'],
-            ] : [
-                'width' => $calculatedWidth,
-                'height' => $calculatedHeight,
-            ];
-        }
-
-        if ($imageSize['width'] >= $imageSize['height']) {
-            $factor = $preferredThumbnailSize->getWidth() / $imageSize['width'];
-
-            $calculatedWidth = $preferredThumbnailSize->getWidth();
-            $calculatedHeight = (int) ceil($imageSize['height'] * $factor);
-
-            $useOriginalSizeInThumbnails = $imageSize['width'] < $calculatedWidth || $imageSize['height'] < $calculatedHeight;
+        $calculator = new ThumbnailSizeCalculator();
```

Comment:
i would prefer to register the calculator as a new service and inject it over the constructor, that way it is easier to reuse and it is easier to change the calculator going forward

Reply:
Hi,

I'm sorry, but we don't work with Symfony or PHPStan. If you like this to be changed you would have to do this by yourself.

Kind regards
Kevin